tools/analysis_tools/analyze_results.py

```diff
@@ -97,6 +97,7 @@ def main():
     outputs['gt_label'] = gt_labels
     outputs['gt_class'] = gt_classes
 
+    outputs = {k: v for k, v in outputs.items() if isinstance(v, list)}
```

Comment:
The output json file of `test.py` includes two kinds of keys. One is the predicted results of every sample, including 'class_scores', 'pred_score', 'pred_label', 'pred_class', and another is the metrics result on the entire dataset, like "accuracy_top-1".
So the problem is we only need those predicted results of every sample. Therefore, better to filter those keys, otherwise, it will still not work if some metric result is also a list.
```suggestion
    need_keys = [
        'filename', 'gt_label', 'gt_class', 'pred_score', 'pred_label',
        'pred_class'
    ]
    outputs = {k: v for k, v in outputs.items() if k in need_keys}
```